tests/js/object-tests.js

```diff
@@ -554,46 +554,55 @@ module.exports = {
         });
     },
 
-    async testNotification() {
-        const realm = new Realm({schema: [schemas.StringOnly]});
+    testNotification: async function () {
```

Comment:
I would really love to see this test get rewritten instead of adding on to the pattern of assigning and reassigning the resolve and reject functions to upper scope local variables. It should be pretty straight forward to rewrite into three `new Promise` calls each registering a listener. IMO we're adding to the technical debt here.

Reply:
this PR adds nothing to the test. only fixes it to run successfully on CI. We can definitely improve it but this should be don in another PR.